### PR TITLE
fix for signed/unsigned comparison

### PR DESCRIPTION
**Pull request by hcab14**
compilation error with gcc-8:

```
/home/pi/stlink/src/usb.c: In function '_stlink_usb_version':
/home/pi/stlink/src/usb.c:192:18: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
         if (size != rep_len) {
                  ^~
cc1: all warnings being treated as errors
```

### CODE REVIEW

File: include/stm32.h [hunk 1]
```diff
@@ -9,7 +9,6 @@
 
 // cortex core ids
 #define STM32VL_CORE_ID 0x1ba01477
-#define CS32VL_CORE_ID 0x2ba01477
```
Comment (Nightwalker-87) [inline include/stm32.h]: This has been rolled back in a previous commit.

File: src/flash_loader.c [hunk 2]
```diff
@@ -267,7 +267,6 @@ int stlink_flash_loader_write_to_sram(stlink_t *sl, stm32_addr_t* addr, size_t*
         loader_code = loader_code_stm32l;
         loader_size = sizeof(loader_code_stm32l);
     } else if (sl->core_id == STM32VL_CORE_ID ||
-               sl->core_id == CS32VL_CORE_ID ||
```
Comment (Nightwalker-87) [inline src/flash_loader.c]: This has been rolled back in a previous commit.

File: src/usb.c [hunk 3]
```diff
@@ -189,7 +189,7 @@ int _stlink_usb_version(stlink_t *sl) {
         cmd[i++] = STLINK_APIV3_GET_VERSION_EX;
 
         size = send_recv(slu, 1, cmd, slu->cmd_len, data, rep_len);
-        if (size != rep_len) {
+        if (size == -1 || (uint32_t)(size) != rep_len) {
```
Comment (Nightwalker-87) [inline src/usb.c]: Please check if it works with `size != (ssize_t)rep_len` which is the current state in the develop branch.

Comment (hcab14) [reply src/usb.c]: It does work with the develop branch. Sorry I should have read the README.md file more closely.

Thanks for your review.

Closing this PR

Comment (Nightwalker-87) [reply src/usb.c]: No problem. Thx for your attentiveness and addressing this issue though.

**Pull request closed without merging**